pkg/client/hmc/hmc.go

```diff
@@ -0,0 +1,152 @@
+package hmc
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/ssh"
+	"os"
+	"strings"
+
+	"github.com/ppc64le-cloud/hypershift-agent-automation/util"
+)
+
+type VIOS struct {
+	SSHClient *ssh.Client
+	IP        string
+	UserName  string
+	Password  string
+	HomeDir   string
+}
+
+type Client struct {
+	SSHClient *ssh.Client
+	VIOS      *VIOS
+	IP        string
+	UserName  string
+	Password  string
+}
+
+func LoadEnv(c *Client) error {
+	var set bool
+	var errs []error
+
+	c.IP, set = os.LookupEnv("HMC_IP")
+	if !set {
+		errs = append(errs, fmt.Errorf("HMC_IP env var not set"))
+	}
+	c.UserName, set = os.LookupEnv("HMC_USERNAME")
+	if !set {
+		errs = append(errs, fmt.Errorf("HMC_USERNAME env var not set"))
+	}
+	c.Password, set = os.LookupEnv("HMC_PASSWORD")
+	if !set {
+		errs = append(errs, fmt.Errorf("HMC_PASSWORD env var not set"))
+	}
+	c.VIOS.IP, set = os.LookupEnv("VIOS_IP")
+	if !set {
+		errs = append(errs, fmt.Errorf("VIOS_IP env var not set"))
+	}
+	c.VIOS.UserName, set = os.LookupEnv("VIOS_USERNAME")
+	if !set {
+		errs = append(errs, fmt.Errorf("VIOS_USERNAME env var not set"))
+	}
+	c.VIOS.Password, set = os.LookupEnv("VIOS_PASSWORD")
+	if !set {
+		errs = append(errs, fmt.Errorf("VIOS_PASSWORD env var not set"))
+	}
+	c.VIOS.HomeDir, set = os.LookupEnv("VIOS_HOMEDIR")
+	if !set {
+		errs = append(errs, fmt.Errorf("VIOS_HOMEDIR env var not set"))
+	}
+
+	return errors.Join(errs...)
+}
+
+func NewClient() (*Client, error) {
+	c := &Client{VIOS: &VIOS{}}
+	if err := LoadEnv(c); err != nil {
+		return nil, fmt.Errorf("error reading env for hmc client: %v", err)
+	}
+
+	var err error
+	c.SSHClient, err = util.CreateSSHClient(c.IP, c.UserName, c.Password)
+	if err != nil {
+		return nil, fmt.Errorf("error create hmc ssh client: %v", err)
+	}
+
+	c.VIOS.SSHClient, err = util.CreateSSHClient(c.VIOS.IP, c.VIOS.UserName, c.VIOS.Password)
+	if err != nil {
+		return nil, fmt.Errorf("error create vios ssh client: %v", err)
+	}
+
+	return c, nil
+}
+
+func (hmc Client) GetLPARID(host, lparName string) (string, error) {
+	lparIDCommand := fmt.Sprintf("lshwres -m %s -r virtualio --rsubtype scsi --filter \"lpar_names=%s\"", host, lparName)
+	out, _, err := util.ExecuteRemoteCommand(hmc.SSHClient, lparIDCommand)
+	if err != nil {
+		return "", fmt.Errorf("error executing command to retrieve lpar_id %v", err)
```

Comment:
Sometimes it would be helpful if we can log the actual command which failed to get executed.